--- libol-0.2.24/configure.in~	Sun Aug 11 22:31:45 2002
+++ libol-0.2.24/configure.in	Sun Aug 11 22:31:45 2002
@@ -96,7 +96,5 @@
 	AC_CHECK_LIB(nsl, gethostbyname, LIBS="$LIBS -lnsl",) ) 
 AC_CHECK_LIB(socket, setsockopt)
 AC_CHECK_FUNCS(select strerror syslog)
-AC_CHECK_FUNCS(poll,,[LIBOBJS="jpoll.lo $LIBOBJS"])
-dnl LIBOBJS="jpoll.lo $LIBOBJS"
-AC_SUBST(LIBOBJS)
+AC_CHECK_FUNCS(poll, [],[AC_LIBOBJ([jpoll])])
 
