diff -Nru libol-0.2.9/configure.in libol-0.2.9.new/configure.in
--- libol-0.2.9/configure.in	Wed Dec 15 10:54:01 1999
+++ libol-0.2.9.new/configure.in	Sun Feb 20 09:14:04 2000
@@ -80,12 +80,6 @@
 dnl LIBOBJS="jpoll.lo $LIBOBJS"
 AC_SUBST(LIBOBJS)
 
-if test $with_debug = "yes"; then
-	CFLAGS="-Wall -g"
-else
-	CFLAGS="-Wall -O2"
-fi
-
 MAKE_CLASS="$srcdir/utils/make_class"
 AC_SUBST(MAKE_CLASS)
 AC_SUBST(SCSH)
