--- libol-0.2.24/src/Makefile.am~	Sun Aug 11 22:32:37 2002
+++ libol-0.2.24/src/Makefile.am	Sun Aug 11 22:32:37 2002
@@ -36,7 +36,7 @@
 	pkt_buffer.c resource.c
 # combinators.c io_commands.c command.c
 
-libol_la_LIBADD = $(LIBOBJS)
+libol_la_LIBADD = $(AC_LIBOBJ)
 
 EXTRA_DIST = olconfig.h.in jpoll.c
 
--- libol-0.2.24/configure.in~	Sun Aug 11 22:31:45 2002
+++ libol-0.2.24/configure.in	Sun Aug 11 22:31:45 2002
@@ -96,7 +96,5 @@
 	AC_CHECK_LIB(nsl, gethostbyname, LIBS="$LIBS -lnsl",) ) 
 AC_CHECK_LIB(socket, setsockopt)
 AC_CHECK_FUNCS(select strerror syslog)
-AC_CHECK_FUNCS(poll,,[LIBOBJS="jpoll.lo $LIBOBJS"])
-dnl LIBOBJS="jpoll.lo $LIBOBJS"
-AC_SUBST(LIBOBJS)
+AC_CHECK_FUNCS(poll, [],[AC_LIBOBJ="jpoll.lo $AC_LIBOBJ"])
 
